### DISCUSSION TITLE
no H264 key frames received in 10s, stream can't be decoded

### DISCUSSION THREAD

**Discussion by flatsiedatsie** [bluenviron/mediamtx#2026] [Questions] [answered]
When I run metiamtx with my Raspberry Pi camera I get an endless stream like this:

```
2023/07/05 19:47:16 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:47:27 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:47:39 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:47:50 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:48:01 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:48:13 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:48:24 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:48:35 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:48:47 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:48:58 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:49:09 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:49:21 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:49:32 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:49:43 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:49:54 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:50:06 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
2023/07/05 19:50:17 WAR [path pi] no H264 key frames received in 10s, stream can't be decoded
```

The stream is also quite slow to start?

For example, I'm piping the stream into Homebridge so it shows up as a Homekit camera. About 15 to 20 seconds pass between pressing the play button on my iPhone and viewing the first real image. Sometimes it just conks outl.

I know that libcamera-vid allows for setting the keyframe frequency on h264 stream, perhaps that frequency could be made configurable in the config file?

Device: Raspberry Pi 4 with 4Gb
Camera: Arducam

Unrelated discussion, but also about missing keyframes from an outsides stream:
https://github.com/bluenviron/mediamtx/discussions/1236

**Comment by flatsiedatsie** [comment]
I solved this by lowering the framerate to 640x480. It now only takes about 3 seconds.

**Comment by aler9** [comment, marked as answer]
> I know that libcamera-vid allows for setting the keyframe frequency on h264 stream, perhaps that frequency could be made configurable in the config file?

there's a setting called `rpiCameraIDRPeriod`:

https://github.com/bluenviron/mediamtx/blob/625e7da91d3991f366283597ab5002290811449e/mediamtx.yml#L377

[a comment by an automated account is not shown]
